plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPropertyPage.java

```diff
@@ -202,7 +202,7 @@ private void evaluateFacetConfiguration() {
   private void updatePageBasedOnFacetConfig(boolean hasCorrectFacetConfiguration) {
     if (hasCorrectFacetConfiguration) {
       canSetMessage = true;
-      setErrorMessage(null);
+      content.getDataBindingContext().updateTargets();
```

Comment:
I think it's safe to call `setErrorMessage(null)` first as originally. I remember @nbashirbello was trying to fix #783 with #880. During the course, a subtle bug surfaced (https://github.com/GoogleCloudPlatform/google-cloud-eclipse/pull/880#issuecomment-256149028), and @nbashirbello had to put quite some effort to fix that bug. (For example, `canSetMessage` and its related logic were introduced after the discovery of that subtle bug.) Her intention with `setErrorMessage(null)` was probably to clear the error message set below when switching between the standard preference page and the flex page back and forth. We need to be careful not to make a regression on #783.